### PR TITLE
Fix broken links.

### PR DESCRIPTION
**Pull request by gtback**
I found these using [this Python library](https://wummel.github.io/linkchecker/), on both the public stixproject.github.io site, and verified on a local cloned copy of the site.

There are still a few broken links where STIX and CybOX have the same
type name, but those will be more work to resolve.

### CODE REVIEW

File: documentation/index.md [hunk 1]
```diff
@@ -6,12 +6,13 @@ title: Documentation
 <div class="row">
   <div class="col-md-9">
     <div class="well">
-      <h4><a href="usecases">Use Cases</a></h4>
+      <h4><a href="/usecases">Use Cases</a></h4>
```
Comment (bschmoker) [inline documentation/index.md]: for some reason it works even w/o the leading slash, good to make unambiguous tho

File: documentation/profiles/index.md [hunk 2]
```diff
@@ -141,4 +141,4 @@ STIX profiles are generally created through community agreement (for community p
 
 ## Questions?
 
-Profiles are a complicated topic. Please [get in touch](stix-taxii@hq.dhs.gov) with the team if you have any questions or think this documentation could be improved.
+Profiles are a complicated topic. Please [get in touch](mailto:stix-taxii@hq.dhs.gov) with the team if you have any questions or think this documentation could be improved.
```
Comment (bschmoker) [inline documentation/profiles/index.md]: @gtback is this email live?

File: getting-started/analyst-exercise/index.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ active: getting-started
 ---
 
 ## Setup
-Consult this [FireEye report](report) for a prose description of malicious actors from Iran targeting governments and commercial computers.
+Consult this [FireEye report](report.html) for a prose description of malicious actors from Iran targeting governments and commercial computers.
```
Comment (bschmoker) [inline getting-started/analyst-exercise/index.md]: at least in chrome the trialing file extension seems optional

**Comment by bschmoker** [comment]
looks good to me @gtback - making note to include in future CI configs :smile_cat:

File: documentation/index.md (hunk 1 above)
Comment (gtback) [inline documentation/index.md]: In FF, if I try to go to http://stixproject.github.io/documentation, it redirects me to http://stixproject.github.io/documentation/ (add the trailing slash), so the relative link "usecases" goes to http://stixproject.github.io/documentation/usecases, which doesn't exist. The "Go" button already worked; this was for the heading above it.

File: documentation/profiles/index.md (hunk 2 above)
Comment (gtback) [inline documentation/profiles/index.md]: I don't know. The address on home page got changed back to stix@mitre.org at some point, but this didn't get changed. I'll let someone else decide what the link should actually be :smile:

Comment (johnwunder) [inline documentation/profiles/index.md]: We probably should make it a configuration option and set it to `stix@mitre.org` for now. I want to make it configuration bc I think we might switch it to the cti-users list at some point.

File: getting-started/analyst-exercise/index.md (hunk 3 above)
Comment (gtback) [inline getting-started/analyst-exercise/index.md]: Hmmm, they are both working for me now (at least on stixproject.github.io). Without the .html doesn't work for me locally. I wonder if GitHub doesn't remove old pages.

**Comment by gtback** [comment]
Thanks, @bschmoker

File: documentation/profiles/index.md (hunk 2 above)
Comment (bschmoker) [inline documentation/profiles/index.md]: @johnwunder - I'll make a PR today if that's cool w/ you

Comment (johnwunder) [inline documentation/profiles/index.md]: That would be awesome, thanks!

Comment (bschmoker) [inline documentation/profiles/index.md]: https://github.com/STIXProject/stixproject.github.io/pull/276

**Pull request merged**